### PR TITLE
[ADPF] PerformanceHintAPI - include core threads + use NDK API

### PR DESCRIPTION
**Pull request by HakimHauston**
Re: #

adapt PerformanceHintAPI to include main thread & render thread and optionally use the NDK API when available

### Changelog

* Add call to ADPF Performance Hint Manager in MessageQueue
* Adapt ADPFManager to be able to dynamically add/remove thread-ids

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ V] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/base/threading/MessageQueue.cpp
Comment (minggo) [inline native/cocos/base/threading/MessageQueue.cpp]: MessageQueue is used for all platforms, and this header file is not valid for all platforms.

File: native/cocos/base/threading/MessageQueue.cpp
Comment (minggo) [inline native/cocos/base/threading/MessageQueue.cpp]: MessageQueue is used for all platforms and currently ADPF is only available on Android.

File: native/cocos/base/threading/MessageQueue.cpp
Comment (HakimHauston) [reply native/cocos/base/threading/MessageQueue.cpp]: Thanks, made the adjustment in 23d0327

File: native/cocos/base/threading/MessageQueue.cpp
Comment (HakimHauston) [reply native/cocos/base/threading/MessageQueue.cpp]: Thanks, made the adjustment in 23d0327

**Comment by minggo** [comment]
Please refer the compiling error on Android
```
 /home/runner/work/cocos-engine/cocos-engine/native/cocos/base/threading/MessageQueue.cpp:284:5: error: use of undeclared identifier 'ADPFManager'
      ADPFManager::getInstance().AddThreadIdToHintSession(tid);
```

File: native/cocos/platform/android/adpf_manager.cpp
Comment (minggo) [inline native/cocos/platform/android/adpf_manager.cpp]: It is better to modify the format as follow to unify the coding style as other codes in the same file.
```c++
void ADPFManager::AddThreadIdToHintSession(int32_t tid) {
```

Comment (HakimNB) [reply native/cocos/platform/android/adpf_manager.cpp]: Thanks, modified manually in `7a3a44b103`, my `clang-tidy` setup isn't working as expected, the autofix remove essential #includes accidentally

**Comment by minggo** [comment]
Could you please fix the clang tidy error?

File: native/cocos/platform/android/adpf_manager.cpp
Comment (minggo) [inline native/cocos/platform/android/adpf_manager.cpp]: Function name should start with lower case as other function names.

File: native/cocos/platform/android/adpf_manager.cpp
Comment (minggo) [inline native/cocos/platform/android/adpf_manager.cpp]: The same as above.

File: native/cocos/platform/android/adpf_manager.cpp
Comment (minggo) [reply native/cocos/platform/android/adpf_manager.cpp]: Or it should have the same rule in this file.

File: native/cocos/platform/android/adpf_manager.cpp
Comment (minggo) [inline native/cocos/platform/android/adpf_manager.cpp:320]: Other function names start with upper case, but this function starts with lower case. Please use the same rule.

File: native/cocos/platform/android/adpf_manager.cpp
Comment (minggo) [inline native/cocos/platform/android/adpf_manager.cpp]: It is better to reduce the spaces.

File: native/cocos/platform/android/adpf_manager.cpp
Comment (minggo) [inline native/cocos/platform/android/adpf_manager.cpp]: It seems data is not used.

File: native/cocos/platform/android/adpf_manager.cpp
Comment (minggo) [inline native/cocos/platform/android/adpf_manager.cpp]: The same as above.

File: native/cocos/platform/android/adpf_manager.cpp
Comment (minggo) [inline native/cocos/platform/android/adpf_manager.cpp:315]: The usage of std::move() depends on c++ 20?

File: native/cocos/platform/android/adpf_manager.cpp
Comment (minggo) [inline native/cocos/platform/android/adpf_manager.cpp]: It seems result is only used in `CC_LOG_INFO `?

File: native/cocos/platform/android/adpf_manager.h
Comment (minggo) [inline native/cocos/platform/android/adpf_manager.h]: It is better to use `std::unordered_map<>` if the order is not important.

File: native/cocos/platform/android/adpf_manager.cpp
Comment (HakimHauston) [reply native/cocos/platform/android/adpf_manager.cpp:315]: Should be available since the ANSI C++, it is the one from <algorithm> https://[cplusplus.com/reference/algorithm/remove/](https://cplusplus.com/reference/algorithm/remove/)

Comment (minggo) [reply native/cocos/platform/android/adpf_manager.cpp:315]: It is wired. What i referred is the doc: https://en.cppreference.com/w/cpp/algorithm/remove. It seems c++17 only supports
```c++
template< class ExecutionPolicy, class ForwardIt, class T >
ForwardIt remove( ExecutionPolicy&& policy,
                  ForwardIt first, ForwardIt last, const T& value );
```

Comment (minggo) [reply native/cocos/platform/android/adpf_manager.cpp:315]: Sorry, i missed the `until c++20` for 
```c++
template< class ForwardIt, class T >
ForwardIt remove( ForwardIt first, ForwardIt last, const T& value );
```

File: native/cocos/platform/android/adpf_manager.cpp
Comment (HakimHauston) [reply native/cocos/platform/android/adpf_manager.cpp]: Changed all to lower case in bfc213811f

File: native/cocos/platform/android/adpf_manager.cpp
Comment (HakimHauston) [reply native/cocos/platform/android/adpf_manager.cpp]: standardized in 446d85bed2

File: native/cocos/platform/android/adpf_manager.cpp
Comment (HakimHauston) [reply native/cocos/platform/android/adpf_manager.cpp:320]: Changed all to lower case in https://github.com/cocos/cocos-engine/commit/bfc213811fcea4240c7b82294c0552f7eed2fdef

File: native/cocos/platform/android/adpf_manager.cpp
Comment (HakimHauston) [reply native/cocos/platform/android/adpf_manager.cpp]: Reduced in 446d85bed2

File: native/cocos/platform/android/adpf_manager.cpp
Comment (HakimHauston) [reply native/cocos/platform/android/adpf_manager.cpp]: Cleaned up in 1d6c8b9359

File: native/cocos/platform/android/adpf_manager.cpp
Comment (HakimHauston) [reply native/cocos/platform/android/adpf_manager.cpp]: Cleaned up data in 1d6c8b9359

File: native/cocos/platform/android/adpf_manager.cpp
Comment (HakimHauston) [reply native/cocos/platform/android/adpf_manager.cpp]: Cleaned up variables in 1d6c8b9359

File: native/cocos/platform/android/adpf_manager.h
Comment (HakimHauston) [reply native/cocos/platform/android/adpf_manager.h]: Thanks, order isn't important, only the key value mapping is, changed in 1d6c8b9359

**Review by minggo — APPROVED** [review]

**Comment by minggo** [comment]
@HakimHauston thanks, i will merge it after @HakimNB approve it.

**Comment by HakimNB** [comment]
> @HakimHauston thanks, i will merge it after @HakimNB approve it.

Thanks.. but I am not authorized to approve it..?

**Comment by minggo** [comment]
@HakimNB i am not sure, i thought everyone had the permission. But no matter, you can reply in comment if you approve it.

**Comment by HakimNB** [comment]
@minggo I approve this PR, feel free to merge it anytime you deem fit

**Review by HakimNB — APPROVED** [review]
@minggo approved

**Pull request merged**
